ddl/ddl_api.go

```diff
@@ -2762,8 +2762,10 @@ func checkTwoRangeColumns(ctx sessionctx.Context, curr, prev *model.PartitionDef
 	for i := 0; i < len(pi.Columns); i++ {
 		// Special handling for MAXVALUE.
 		if strings.EqualFold(curr.LessThan[i], partitionMaxValue) {
-			// If current is maxvalue, it certainly >= previous.
-			return true, nil
+			if !strings.EqualFold(prev.LessThan[i], partitionMaxValue) {
```

Comment:
I think combine 2764 and 2765 in one line is better.

Reply:
@Defined2014 Exactly, fixed.